src/Compilers/CSharp/Portable/Errors/ErrorFacts.cs

```diff
@@ -198,8 +198,8 @@ internal static int GetWarningLevel(ErrorCode code)
         {
             if (IsInfo(code) || IsHidden(code))
             {
-                // Info and hidden diagnostics have default warning level.
-                return Diagnostic.DefaultWarningLevel;
+                // Info and hidden diagnostics should always be produced because some analyzers depend on them.
+                return 1;
```

Comment:
Should this return 0 instead?

Reply:
Changing both info and hidden seems fine to me, as info doesn't affect the command line build.

---
In reply to: [501191632](https://github.com/dotnet/roslyn/pull/48397#discussion_r501191632) [](ancestors = 501191632)

Reply:
Rather than use a literal here I think we should use a named constant. I would add one next to `DefaultWarningLevel` with a doc comment summarizing what it represents

Reply:
@jaredpar Done